### PR TITLE
change parse_format function

### PR DESCRIPTION
**Pull request by agila5**
I change the `parse_format` function according to what was discussed in the issue https://github.com/r-lib/scales/issues/203. Now the same code retuns:

``` r
library(ggplot2)

# data
fake_data <- data.frame(
  classes = c("alpha", "beta"),
  value = c(1, 2),
  stringsAsFactors = FALSE
)

ggplot(fake_data) + 
  geom_col(aes(x = classes, y = value, fill = classes)) + 
  scale_fill_discrete(labels = scales::parse_format) + 
  scale_x_discrete(labels = scales::parse_format)
```

<sup>Created on 2019-04-11 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

The strange thing is that the same code returns a different plot (i.e. the fill labels are too close to the colours) if I view it using Rstudio: 
Rplot

### CODE REVIEW

File: R/formatter.r [hunk 1]
```diff
@@ -588,18 +588,28 @@ ordinal_spanish <- function() {
...
+#'
 #' @export
 #' @examples
-#' parse_format()(c("alpha", "beta", "gamma"))
-parse_format <- function() {
-  function(x) {
-    lapply(as.character(x), function(x) parse(text = x, srcfile = NULL))
+#' parse_format(c("alpha", "beta", "gamma"))
+#' parse_format(1:5)
+parse_format <- function(text) {
```
Comment (hadley) [inline R/formatter.r]: You need to preserve the original structure of this function you code should look like:

```R
parse_format <- function() {
  function(x) {
     x <- as.character(x) 
     out <- ...
     ...
  }
}
```

**Comment by hadley** [comment]
Thanks!

For future reference, if you include `Fixes #xyz` in a commit message (or the text of the PR), the corresponding issue will be automatically closed when the PR is merged.

**Comment by agila5** [comment]
> For future reference, if you include Fixes #xyz in a commit message (or the text of the PR), the corresponding issue will be automatically closed when the PR is merged.

Ok, thanks. I always forget about that. 

Anyway, can you reproduce the problem with the labels positioning in the legend? I still get the same plot as in the first post of this conversation where, if I view the plot using Rstudio, the labels, in my opinion, are way too close to the legend. If I use the CRAN version of `scales` (i.e. the old version of `parse_format `), the problem disappears.

**Comment by hadley** [comment]
The code that works for me is

```R
ggplot(fake_data) + 
  geom_col(aes(x = classes, y = value, fill = classes)) + 
  scale_fill_discrete(labels = scales::parse_format()) + 
  scale_x_discrete(labels = scales::parse_format())
```

But if you're having a problem, it would be better to file a new issue, rather than discussing it in a closed PR.

**Pull request merged**